File: Easy/016_Rightmost/C#/Program.cs
using System;
using System.IO;

class Program
{
    /// <summary>
    /// This method prints out the position of the right most char in the string
    /// </summary>
    public static void RightMost(string line)
    {
        // get the last char in the line
        char ch = line[line.Length - 1];
        // initially set pos to -1
        int pos = -1;

        // loop through the string
        for (int i=0; i<line.Length-2; ++i)
        {
            // if the current char is the char, set pos to i
            if (line[i] == ch)
                pos = i;
        }
        // print out the pos (-1 if not found)
        Console.WriteLine(pos);
    }

    public static void Main (string[] args)
    {
        // for each line in the file, Get right most char position
        foreach (var line in File.ReadLines(args[0]))
            RightMost(line);
    }
}

File: Easy/016_Rightmost/Java/Program.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.*;

public class Main
{

    public static void main(String[] args) throws IOException
    {

        if (args.length > 0)
        {
            // create a reader to read the file
            BufferedReader reader = new BufferedReader(new FileReader(args[0]));

            // init a string to hold each line
            String line = "";

            // while able to get a line from the file, add to the list
            while ((line = reader.readLine()) != null)
                rightMost(line);

            // close out the reader
            reader.close();
        }
    }

    // This method prints out the location of the 
    // right most char
    // Input: Hello World,r
    // Ouput: location of 'r' in 'Hello World' : 8
    private static void rightMost(String line) {
        // split the line on the comma
        String data[] = line.split(",");
        // get the char in question 
        char ch = data[1].charAt(0);
        // init a loc to -1, if not found will output -1
        int loc = -1;
        // loop through the string in question
        for (int i = 0; i < data[0].length(); i++) {
            // if the current char is the char in question
            if (data[0].charAt(i) == ch)
                // set the loc to the current
                loc = i;
        }
        // output loc, if not found it will be -1
        System.out.println(loc);
    }

}